lib/galaxy/jobs/runners/local.py

```diff
@@ -49,23 +49,53 @@ def __init__(self, app, nworkers):
 
         super().__init__(app, nworkers)
 
+    def get_local_gpu_count(self):
+        bash_command = "/bin/bash -c 'nvidia-smi --query-gpu=index --format=csv'"
+        sp = subprocess.Popen(bash_command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        out, err = sp.communicate()
+        command_not_found = "command not found"
+        if command_not_found.encode() not in out and command_not_found.encode() not in err:
+            data = (out.decode('utf-8').splitlines())[1:] #eliminate first element, as it is the header
```

Comment:
Please use the python csv module (https://docs.python.org/3/library/csv.html) so that this value can be parsed more robustly.